libraries/AP_Airspeed/AP_Airspeed_MS4525.cpp

```diff
@@ -120,6 +126,24 @@ void AP_Airspeed_MS4525::_collect()
     dT_raw = (data[2] << 8) + data[3];
     dT_raw = (0xFFE0 & dT_raw) >> 5;
 
+    int16_t dp_raw2, dT_raw2;
+    dp_raw2 = (data2[0] << 8) + data2[1];
+    dp_raw2 = 0x3FFF & dp_raw2;
+    dT_raw2 = (data2[2] << 8) + data2[3];
+    dT_raw2 = (0xFFE0 & dT_raw2) >> 5;
+
+    // reject any values that are the absolute minum or maximums
+    // these can happen due to gnd lifts or communication erros on the bus
+    if (dp_raw  == 0x3FFF || dp_raw  == 0 || dT_raw  == 0x7FF || dT_raw == 0 ||
+        dp_raw2 == 0x3FFF || dp_raw2 == 0 || dT_raw2 == 0x7FF || dT_raw2 == 0) {
+        return;
+    }
+
+    // reject any double reads where the value has shifted in the upper more then a byte as corrupted
```

Comment:
The sentence looks like a mix of two. I would change it to:

reject any double reads where the values have a difference bigger than 255 (one lower byte)